src/vs/code/node/cli.ts

```diff
@@ -319,23 +324,75 @@ export async function main(argv: string[]): Promise<any> {
 			options['stdio'] = 'ignore';
 		}
 
-		const child = spawn(process.execPath, argv.slice(2), options);
+		let child: ChildProcess;
+		if (!isMacintosh) {
+			// We spawn process.execPath directly
+			child = spawn(process.execPath, argv.slice(2), options);
+		} else {
+			// On mac, we spawn using the open command to obtain behavior
+			// similar to if the app was launched from the dock
+			// https://github.com/microsoft/vscode/issues/102975
+			const openArgs: string[] = ['-n'];
+
+			if (verbose || hasReadStdinArg || args.wait) {
+				openArgs.push('-W');
+			}
+
+			if (verbose) {
+				// The open command only allows for redirecting stderr and stdout to files,
+				// so we make it redirect those to temp files, and then use a logger to
+				// redirect the file output to the console
+				function createLoggerPromise(logger: CliVerboseLogger, filename: string, stream: NodeJS.WriteStream): (child: ChildProcess) => Promise<void> {
+					return async (child: ChildProcess) => {
+						const childClosePromise = new Promise<void>(resolve => {
+							child.on('close', () => {
+								resolve();
+							});
+						});
+						await Promise.race([logger.streamFile(filename, stream), childClosePromise]);
+						unlinkSync(filename);
```

Comment:
Should this maybe be in a `try/finally` in case the `race` fails for some reason?